sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ChunkedDownloadUtils.java

```diff
@@ -40,7 +40,7 @@
             ? range.getCount() : parallelTransferOptions.getBlockSizeLong();
 
         return downloader.apply(new BlobRange(range.getOffset(), initialChunkSize), requestConditions)
-            .subscribeOn(Schedulers.elastic())
+            .publishOn(Schedulers.boundedElastic())
```

Comment:
@kasobol-msft, @rickle-msft, @gapra-msft, I updated this to use `boundedElastic` over `elastic` as `elastic` has been deprecated and allows for an unbounded number of threads to be created.

Additionally, I changed this from `subscribeOn` to `publishOn` as `subscribeOn` affects the entire reactive chain, including all reactive operations applied before this point in time. `publishOn` only affects the reactive chain after the `publishOn` operator. So, with this chain operators before this point should use either the default or what a customer has configured, after this point it will use the thread-pool based `boundedElastic` scheduler to handle blocking I/O with potentially many threads.